### PR TITLE
SNOW-121291: Adding sample for authentication by key pair from file 

### PR DESCRIPTION
**Pull request by sfc-gh-aling**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-121291

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

**Review by sfc-gh-mkeller — COMMENTED** [review]
Please add a `README.md` to the `samples` directory that describes what these files are for

**Comment by sfc-gh-aling** [comment]
@sfc-gh-mkeller added, let me know how it looks to you

File: samples/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Snowflake Connector for Python Samples
+
+These are code samples that show common scenario operations with the Snowflake Connector for Python library.
+
+- auth_by_key_pair_from_file.py.py - Example to implement a key pair authentication plugin
```
Comment (sfc-gh-mkeller) [inline samples/README.md:5]:
```suggestion
- auth_by_key_pair_from_file.py.py - Example to implement a custom key pair authentication plugin
```

**Review by sfc-gh-mkeller — APPROVED** [review]
:ship: :ship:

File: samples/README.md (hunk 1 above)
Comment (sfc-gh-sfan) [reply samples/README.md:5]: nit: typo? `auth_by_key_pair_from_file.py.py` -> `auth_by_key_pair_from_file.py`

**Review by sfc-gh-sfan — APPROVED** [review]

File: samples/README.md (hunk 1 above)
Comment (sfc-gh-mkeller) [reply samples/README.md:5]: What @sfc-gh-sfan said
```suggestion
- auth_by_key_pair_from_file.py - Example to implement a key pair authentication plugin
```

Comment (sfc-gh-aling) [reply samples/README.md]: updated, thanks

**Pull request merged**